tests/core/test_macros.py

```diff
@@ -754,3 +754,223 @@ def test_deduplicate_error_handling(macro_evaluator):
         str(e.value.__cause__)
         == "order_by must be a list of strings, optional - nulls ordering: ['<column> <asc|desc> nulls <first|last>']"
     )
+
+
+@pytest.mark.parametrize(
+    "dialect, date_part",
```

Comment:
Should we add the expected SQL in this parameterize? Or just parameterize on the date part since we're already checking all dialects in that nested if-elif-else block?